docs/integrations/web.md

```diff
@@ -36,209 +36,177 @@ There are a couple important concepts to grasp for using Armory. We’ll look at
+
+## Creating components and templates
+
+### Custom components
+
+Let's take a look at how you can add new custom components to your screens. Let's say you wanted to add a component for uploading a document to your screen.
+The steps are as follows:
+1. Create the components directory in `services/armory/src/components` [TODO include by default?]
+2. Add the component in vue, for example create the file:
+```vue title="services/armory/src/components/FileUpload.vue"
+<template>
+  <div class="file is-boxed">
+    <label class="file-label">
+      <input class="file-input" type="file" name="resume" />
+      <span class="file-cta">
+        <span class="file-icon">
+          <i class="fas fa-upload"></i>
+        </span>
+        <span class="file-label"> Choose a file… </span>
+      </span>
+    </label>
+  </div>
+</template>
+
+<script>
+import BaseValue from "armory-sdk/src/components/base/BaseValue";
+
+export default {
+  name: "FileUpload",
+  extends: BaseValue,
+  props: {
+    placeholder: {
+      type: String,
+      default: "",
+    },
+  },
+};
+</script>
+
+<style></style>
+```
 
+:::tip
+The HTML you see in the `<template>...</template>` node was taken from this [website](https://bulma.io/documentation/). There are plenty of such resources out there, if you're not too crafty with HTML yourself!
+:::
 
-```java title="rule_engine/src/main/resources/rules/mindy/Mindy.drl"
+3. Finally, add your custom component to the `App.vue` file:
+```java title="services/armory/src/App.vue"
+...
+import FileUpload from "./components/FileUpload.vue";
 ...
+    getCustomComponents() {
+      return {FileUpload};
+...
+```
 
-rule "Start user onboarding"
-    when
-        signal: Submit(buttonId == "nameSubmited") from entry-point "signals"
-        agent: Felix()
-    then
-        modify(agent){
-            setName(signal.getParamAsString("name"))
-            };
-        agent.showOnboardingScreens();
-        delete(signal);
-end
+You can now just use your custom component directly in Java using the `CustomComponent` class:
+```java
+new Screen("uploadDocument")
+    .add(new CustomComponent("FileUpload"))
 ```
+In case the component needs some parameters, those are simply added as key-value pairs to the `CustomComponent` params field. For example, this is how you would fill in the data for the component consisting of some label and a description that goes with it: `new CustomComponent("LabeledDescription").addParam("label", "Attendees:").addParam("description", "There will be 270 attenedees at the event.")`).
```

Comment:
I suggest you have this code as a new code block. Also, you can easily do that for `FileUpload` since it allows for a `placeholder` parameter.